boards/kernel_layout.ld

```diff
@@ -337,6 +337,40 @@ SECTIONS
     } > ram
     _eappmem = ORIGIN(ram) + LENGTH(ram);
 
+
+    /* Place attributes at the end of the kernel ROM region. These will be used
+     * for host-side tools to learn about the installed kernel.
+     *
+     * Attributes are stored as TLVs, but going backwards in flash.
+     *
+     *                           2 byte   2 byte   4 byte
+     * -----------+------------+--------+--------+------------+
+     *            | Value...   | Type   | Length | TOCK       | Start of `prog`
+     * -----------+------------+--------+--------+------------+
+     * <-TLVs...-> <-----TLV--------------------> <-Sentinel->
+     *
+     * `Length` specifies the size of the value (i.e. not include the header).
+     */
+    .attributes :
+    {
+        . = ABSOLUTE(ORIGIN(rom) + LENGTH(rom) - 16);
+
+        /* TLV: App Memory
+         * This indicates the start address and size of RAM available for
+         * userspace apps.
+         */
+        LONG(_sappmem) /* Address of start of app memory. */
+        LONG(_eappmem - _sappmem) /* Length of app memory. */
+        SHORT(0x0101) /* Type = App RAM Region = 0x0101 */
```

Comment:
Yeah these are not in apps so they are different, but given that the numbers don't actually matter, it seems prudent to just choose different numbers to avoid potential confusion.

Reply:
I'm happy to document this should we choose to go forward with it.